packages/timeline-state-resolver-types/src/tricaster.ts

```diff
@@ -0,0 +1,307 @@
+	| MappingTriCasterMixEffect
+	| MappingTriCasterDownStreamKeyer
+	| MappingTriCasterInput
+	| MappingTriCasterAudioChannel
+	| MappingTriCasterMixOutput
+
+export interface TriCasterOptions {
+	host: string
+	port: number
+}
+
+export enum TimelineContentTypeTriCaster {
+	ME = 'ME',
+	DSK = 'DSK',
+	AUDIO_CHANNEL = 'AUDIO_CHANNEL',
+	INPUT = 'INPUT',
+	MIX_OUTPUT = 'MIX_OUTPUT',
+}
+
+export type TimelineObjTriCasterAny =
+	| TimelineObjTriCasterME
+	| TimelineObjTriCasterDSK
+	| TimelineObjTriCasterInput
+	| TimelineObjTriCasterAudioChannel
+	| TimelineObjTriCasterMixOutput
+
+export interface TimelineObjTriCasterBase extends TSRTimelineObjBase {
+	content: {
+		deviceType: DeviceType.TRICASTER
+		type: TimelineContentTypeTriCaster
+	} & TimelineDatastoreReferencesContent
+}
+
+export type TriCasterMixEffect = {
+	/** Discarded when layers are defined (M/E in effect mode) */
+	programInput?: string
+
+	/** Discarded when transition other than 'cut' is used */
+	previewInput?: string
+
+	transition?: TriCasterTransition
+
+	keyers?: Record<TriCasterKeyerName, TriCasterKeyer>
+
+	/** Use only in conjunction with effects that use M/E rows as layers (e.g. LiveSets) */
+	layers?: Partial<Record<TriCasterLayerName, TriCasterLayer>>
+
+	/** Default: 'background' */
+	delegates?: TriCasterDelegateName[]
+}
+
+export interface TimelineObjTriCasterME extends TimelineObjTriCasterBase {
+	content: {
+		deviceType: DeviceType.TRICASTER
+		type: TimelineContentTypeTriCaster.ME
+
+		me: TriCasterMixEffect
+	} & TimelineDatastoreReferencesContent
+}
+
+export function isTimelineObjTriCasterME(timelineObject: TSRTimelineObjBase): timelineObject is TimelineObjTriCasterME {
+	return (timelineObject as TimelineObjTriCasterBase).content?.type === TimelineContentTypeTriCaster.ME
+}
+
+/**
+ * Convenience object for the keyers in the Main M/E
+ */
+export interface TimelineObjTriCasterDSK extends TimelineObjTriCasterBase {
+	content: {
+		deviceType: DeviceType.TRICASTER
+		type: TimelineContentTypeTriCaster.DSK
+
+		keyer: TriCasterKeyer
+	} & TimelineDatastoreReferencesContent
+}
+
+export function isTimelineObjTriCasterDSK(
+	timelineObject: TSRTimelineObjBase
+): timelineObject is TimelineObjTriCasterDSK {
+	return (timelineObject as TimelineObjTriCasterBase).content?.type === TimelineContentTypeTriCaster.DSK
+}
+
+export interface TriCasterInput {
+	videoActAsAlpha?: boolean
+	videoSource?: string
+}
+
+export interface TimelineObjTriCasterInput extends TimelineObjTriCasterBase {
+	content: {
+		deviceType: DeviceType.TRICASTER
+		type: TimelineContentTypeTriCaster.INPUT
+
+		input: TriCasterInput
+	} & TimelineDatastoreReferencesContent
+}
+
+export function isTimelineObjTriCasterInput(
+	timelineObject: TSRTimelineObjBase
+): timelineObject is TimelineObjTriCasterInput {
+	return (timelineObject as TimelineObjTriCasterBase).content?.type === TimelineContentTypeTriCaster.INPUT
+}
+
+export interface TriCasterAudioChannel {
+	isMuted?: boolean
```

Comment:
Does this really need to be optional? If it's undefined then it's just the same as false

Reply:
If it's undefined then it might be provided by another timeline object on another layer targetting the same channel, see https://github.com/tv2/tv-automation-state-timeline-resolver/pull/46#discussion_r1084981197. Let's talk about it if you'd like.